src/trainingdata/trainingdata.h

```diff
@@ -90,28 +87,28 @@ static_assert(sizeof(V6TrainingData) == 8356, "Wrong struct size");
 
 #pragma pack(pop)
 
-class TrainingDataWriter {
+class V6TrainingDataArray {
  public:
-  // Creates a new file to write in data directory. It will has @game_id
-  // somewhere in the filename.
-  TrainingDataWriter(int game_id);
-
-  ~TrainingDataWriter() {
-    if (fout_) Finalize();
-  }
-
-  // Writes a chunk.
-  void WriteChunk(const V6TrainingData& data);
-
-  // Flushes file and closes it.
-  void Finalize();
-
-  // Gets full filename of the file written.
-  std::string GetFileName() const { return filename_; }
+  V6TrainingDataArray(FillEmptyHistory white_fill_empty_history,
+                      FillEmptyHistory black_fill_empty_history,
+                      pblczero::NetworkFormat::InputFormat white_input_format,
+                      pblczero::NetworkFormat::InputFormat black_input_format)
```

Comment:
I would suggest only taking in one input_format - makes it clearer why we're checking and asserting that they are the same back near the call site.